lms/djangoapps/courseware/model_data.py

```diff
@@ -338,10 +354,16 @@ def get(self, key):
         if key.scope not in self._allowed_scopes:
             raise InvalidScopeError(key)
 
+        if self._from_django_cache(key):
+            value = DJANGO_FIELD_CACHE.get(self._django_cache_key(key))
+            if value is None:
+                raise KeyError(key.field_name)
```

Comment:
This is confusing: you look up a value in the DJANGO_FIELD_CACHE, and if it isn't None, you never use it for anything?  I suspect there's a good reason for this, but I don't know what it is.